src/components/CompareResults/RetriggerButton.tsx

```diff
@@ -1,37 +1,126 @@
 import {
   getTaskclusterCredentials,
   signInIntoTaskcluster,
 } from '../../logic/taskcluster';
 import { Strings } from '../../resources/Strings';
 
+type CenteredModalProps = {
+  open: boolean;
+  onClose: () => unknown;
+  ariaLabelledby: string;
+  children: React.ReactNode;
+  paperStyle?: React.ComponentProps<typeof Paper>['sx'];
+};
+
+function CenteredModal(props: CenteredModalProps) {
+  return (
+    <Modal
+      open={props.open}
+      onClose={props.onClose}
+      aria-labelledby={props.ariaLabelledby}
+      sx={{ display: 'flex' }}
+    >
+      <Paper
+        sx={{
+          margin: 'auto',
+          padding: 6,
+          maxHeight: 0.6,
+          width: 0.6,
+          maxWidth: 500,
+          overflow: 'auto',
+          position: 'relative' /* to be able to position the close icon */,
+          ...props.paperStyle,
+        }}
+      >
+        <IconButton
+          aria-label='Close'
+          size='large'
+          sx={{ position: 'absolute', top: 8, right: 8 }}
+          onClick={props.onClose}
+        >
+          <CloseIcon />
+        </IconButton>
+        {props.children}
+      </Paper>
+    </Modal>
+  );
+}
+
 type SignInModalProps = {
   open: boolean;
   onClose: () => unknown;
   onSignIn: () => unknown;
 };
-function SignInModal({ open, onClose, onSignIn }: SignInModalProps) {
+function SignInModal(props: SignInModalProps) {
+  const [waitingSignIn, setWaitingSignIn] = useState(false);
   const onSignInButtonClick = () => {
-    onSignIn();
+    if (waitingSignIn) {
+      return;
+    }
+    setWaitingSignIn(true);
+    props.onSignIn();
+  };
+
+  const onClose = () => {
+    setWaitingSignIn(false);
+    props.onClose();
   };
+
   return (
-    <Modal
-      open={open}
+    <CenteredModal
+      open={props.open}
       onClose={onClose}
-      aria-labelledby='sign-in-modal-title'
-      sx={{ display: 'flex' }}
+      ariaLabelledby='sign-in-modal-title'
+      paperStyle={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        gap: 2,
+        paddingBottom: 5,
+        textAlign: 'center',
+      }}
     >
-      <Box>
-        <Typography id='sign-in-modal-title' component='h2'>
-          Sign into Taskcluster to re-trigger the comparison
-        </Typography>
-        <Button onClick={onSignInButtonClick}>Sign in</Button>
+      <Box
+        sx={{
+          filter: (theme) =>
+            theme.palette.mode === 'dark' ? 'brightness(2)' : null,
+        }}
+      >
+        <img src={TaskclusterLogo} role='presentation' width='48' />
       </Box>
-    </Modal>
+      <Typography id='sign-in-modal-title' component='h2' variant='h2'>
+        Sign into Taskcluster to re-trigger the comparison
+      </Typography>
+      <Typography>
+        To be able to retrigger a task within Taskcluster, it’s necessary that
+        you log in to it first. Then PerfCompare will retrigger the task on your
+        behalf.
```

Comment:
I remember you mentioned something about getting rid of the `Strings` file in the resources folder but for consistency sake, would you mind putting this text there? If not, we should probably have a discussion on how to handle text in the future.

Reply:
My personal view is that the Strings file is "just another indirection" that serves no useful purpose at the moment.

I was thinking that the Strings file was for localization, but if we want to do localization in the future, we're better off using https://projectfluent.org/ instead of rolling on our own system. In that case having all strings in the Strings file could be of some help so that we could quickly see what we have to extract there.

However I don't think localizing this project would be useful because it's not user-facing but rather internal developer-facing.

In the mean time though I'll do as you suggest for consistency.